src/pages/gd/Stake/Savings/index.tsx

```diff
@@ -16,32 +16,107 @@ import { SavingsDepositMobile } from './SavingsDepositMobile'
 import { HeadingCopy } from 'components/Savings/SavingsCard'
 import { useActiveWeb3React } from 'hooks/useActiveWeb3React'
 import { ModalButton } from 'components/Savings/SavingsModal/ModalButton'
+import { ChainId } from '@sushiswap/sdk'
+import { NETWORK_LABEL } from 'constants/networks'
+import Web3SupportedNetworks from 'components/Web3SupportedNetworks'
 
 const SavingsDeposit = styled.div`
     margin-top: 10px;
 `
 
-export const Savings = ({ requiredChain }: { requiredChain: number }): JSX.Element => {
-    const [isOpen, setIsOpen] = useState(false)
-    const { account, chainId } = useActiveWeb3React()
-    const { stats, error } = useSavingsStats(requiredChain, 10)
-    const { i18n } = useLingui()
-    const { width } = useWindowSize()
-    const isMobile = width ? width <= 768 : undefined
-    const sendData = useSendAnalyticsData()
+const SavingRow: FC<{ chainId: ChainId; showModal: (chain: ChainId) => void }> = ({ chainId, showModal }) => {
+    const { stats, error } = useSavingsStats(chainId, 10)
     const { defaultEnv } = useGetEnvChainId()
-    const g$ = G$(requiredChain, defaultEnv)
+    const g$ = G$(chainId, defaultEnv)
+    const { i18n } = useLingui()
+
+    const onModalButtonPress = useCallback(() => showModal(chainId), [chainId, showModal])
 
     useEffect(() => {
         if (error) {
             console.error('Unable to fetch global stats:', { error })
         }
     }, [error])
 
-    const toggleModal = useCallback(() => {
+    return (
+        <>
+            <tr>
+                <td>
+                    <AsyncTokenIcon
+                        address={g$.address}
+                        chainId={g$.chainId}
+                        className={'block w-5 h-5 mr-2 rounded-lg md:w-10 md:h-10 lg:w-12 lg:h-12'}
+                    />
+                </td>
+                <td>{i18n._(t`G$`)}</td>
```

Comment:
isn't this similar to SavingsDepositMobile ? let's reuse components from there

Reply:
@johnsmith-gooddollar 
here is table layout and in SavingsDepositMobile is flex layout

Reply:
or do you mean that SavingsMobileStat component?

Reply:
done